src/core/qgsrange.h

```diff
@@ -392,12 +394,66 @@ class QgsTemporalRange
       return false;
     }
 
+    /**
+     * Extends the range in place by extending this range out to include an \a other range.
+     * If \a other is empty the range is not changed.
+     * If the range is empty and \a other is not, the range is changed and set to \a other.
+     * \see isEmpty()
+     * \since QGIS 3.12
+     * \return TRUE if the range was extended
+     */
+    bool extend( const QgsTemporalRange<T> &other )
+    {
+      if ( other.isEmpty() )
+      {
+        return false;
+      }
+      else if ( isEmpty() )
+      {
+        mLower = other.begin();
+        mUpper = other.end();
+        mIncludeLower = other.includeBeginning();
+        mIncludeUpper = other.includeEnd();
+        return true;
+      }
+
+      // Both not empty, do some math
+      bool changed { false };
+
+      // Lower
+      if ( begin().isValid() && other.begin() < mLower )
```

Comment:
What about when this.begin is valid yet other.begin is not valid? I think that needs to be handled first

Reply:
Apparently not:
```python
QDate() < QDate(2010, 1, 1) == True
```
this is handled correctly, I'm adding this test case:
```python
        range = QgsDateRange(QDate(2010, 4, 6), QDate())
        self.assertTrue(range.extend(QgsDateRange(QDate(), QDate(2010, 5, 2), True, False)))
        self.assertEqual(range, QgsDateRange(QDate(), QDate(), True, True))
```

Reply:
Ok, but the same applies for the upper bound too -- does the same logic work there too?

Reply:
Yes, that's was the missing bit I've added the missing test and change both the conditions for clarity.